### PR TITLE
add AgentOptimizer test in CI 

### LINKED ISSUE #2353: AgentOptimizer test is not included in the CI.
This test is not included in the CI. Could you please add it to contrib-openai and make the PR from the upstream repo to enable CI?

_Originally posted by @sonichi in https://github.com/microsoft/autogen/pull/2299#discussion_r1559503246_

### PR DESCRIPTION
**Pull request by skzhang1**
## Related issue number

Closes #2353
## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: .github/workflows/contrib-openai.yml [hunk 1]
```diff
@@ -334,3 +334,42 @@ jobs:
...
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install packages and dependencies
+        run: |
+          docker --version
+          python -m pip install --upgrade pip wheel
+          pip install -e .
+          python -c "import autogen"
+          pip install coverage pytest-asyncio
```
Comment (sonichi) [inline .github/workflows/contrib-openai.yml:361]: do we need asyncio?

**Comment by sonichi** [comment]
Please make the PR from the upstream repo.

File: .github/workflows/contrib-openai.yml (hunk 1 above)
Comment (WaelKarkoub) [reply .github/workflows/contrib-openai.yml:361]: `AgentOptimizer` doesn't have any async code, safe to just install `pytest`

**Review by WaelKarkoub — CHANGES_REQUESTED** [review]
I'm going through the `AgentOptimizer` implementation, and it seems like it's using docker to execute code. We might need to add a `--skip-docker` flag for those tests in case it wasn't installed

**Comment by skzhang1** [comment]
This PR will be made from the upstream repo. So close this one for now.

**Comment by skzhang1** [comment]
> `--skip-docker`

Thanks for the suggestion. Actually, the docker is optional. 

> I'm going through the `AgentOptimizer` implementation, and it seems like it's using docker to execute code. We might need to add a `--skip-docker` flag for those tests in case it wasn't installed

Thanks for the suggestion. Docker is not mandatory for AgentOptimizer. It is optional.

**Pull request closed without merging**